pkg/api/dictionaryinfo.go

```diff
@@ -0,0 +1,338 @@
+
+// parse multiple time layouts commonly seen in Fastly APIs
+func parseFastlyTime(s string) (float64, bool) {
+	if s == "" {
+		return 0, false
+	}
+	layouts := []string{
+		time.RFC3339Nano,
+		time.RFC3339,
+		"2006-01-02 15:04:05",       // "2025-10-21 18:48:00"
+		"2006-01-02T15:04:05",       // without zone
+		"2006-01-02 15:04:05Z07:00", // space + zone
+	}
+	for _, layout := range layouts {
+		if t, err := time.Parse(layout, s); err == nil {
+			return float64(t.Unix()), true
+		}
+	}
+	return 0, false
+}
+
+// GET /service/{sid}/version/{v}/dictionary/{did}/item
+func (c *DictionaryInfoCache) listDictionaryItems(ctx context.Context, serviceID string, version int, dictID string) ([]dictItem, error) {
+	u := fmt.Sprintf("https://api.fastly.com/service/%s/version/%d/dictionary/%s/item", serviceID, version, dictID)
+	req, err := c.req(ctx, http.MethodGet, u)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("list dictionary items: %s", resp.Status)
+	}
+	var items []dictItem
+	return items, json.NewDecoder(resp.Body).Decode(&items)
+}
+
+// Refresh queries Fastly APIs and rebuilds the in-memory snapshot.
+func (c *DictionaryInfoCache) Refresh(ctx context.Context) error {
+	if !c.enabled {
+		return nil
+	}
+	var services []svc
+	if c.serviceCache != nil {
+		for _, s := range c.serviceCache.Services() {
+			active := s.Version
+			services = append(services, svc{
+				ID:   s.ID,
+				Name: s.Name,
+				Versions: []struct {
+					Number int  `json:"number"`
+					Active bool `json:"active"`
+				}{
+					{
+						Number: active,
+						Active: active > 0,
+					},
+				},
+			})
+		}
+	} else {
+		var err error
+		services, err = c.listServices(ctx)
+		if err != nil {
+			return err
+		}
+	}
+	var out []metricRow
+	for _, s := range services {
+		active := -1
+		for _, v := range s.Versions {
+			if v.Active {
+				active = v.Number
+				break
+			}
+		}
+		if active < 0 {
+			continue
+		}
+		dicts, err := c.listDictionaries(ctx, s.ID, active)
+		if err != nil {
+			level.Warn(c.logger).Log("during", "list dictionaries", "service", s.ID, "err", err)
+			continue
+		}
+		for _, d := range dicts {
+			info, err := c.getDictionaryInfo(ctx, s.ID, active, d.ID)
+			if err != nil {
+				level.Warn(c.logger).Log("during", "get dictionary info", "service", s.ID, "dictionary", d.ID, "err", err)
+				continue
+			}
+			var ts float64
+			if v, ok := parseFastlyTime(info.LastUpdated); ok {
+				ts = v
+			} else {
+				// fallback: derive from latest item.updated_at (if any)
+				if items, err := c.listDictionaryItems(ctx, s.ID, active, d.ID); err == nil && len(items) > 0 {
+					var maxTS float64
+					for _, it := range items {
+						if v2, ok2 := parseFastlyTime(it.UpdatedAt); ok2 && v2 > maxTS {
+							maxTS = v2
+						}
+					}
+					ts = maxTS // stays 0 if all missing/unparseable
+				}
+			}
```

Comment:
This should be removed. Because dictionaries can have many thousand of times we don't want to iterate over them and request each item one-at-a-time.

if `info.LastUpdated` doesn't exist that means there are no items in the dictionary so we can just use the dictionary updated_at timestamp. The `dict` struct will need to have the updated_at field added.